File: src/LinkedList/Question1/Solution.java

public class Solution {
    public void deleteNode(ListNode node) {
        ListNode nextNode = node.next;
        node.val = nextNode.val;
        if(nextNode != null){
         node.next = nextNode.next;
        }else{
            node.next = null;
        }
    }

    private class ListNode {
        int val;
        ListNode next;

        public ListNode(int val){ this.val = val; }
    }

}

File: src/LinkedList/Question1/Solution.cpp
struct ListNode {
    int val;
    ListNode *next;

};

class Solution {

public:
  void deleteNode(ListNode* node){
    ListNode* nextNode = node->next;
    node->val = nextNode->val;
    if(nextNode != NULL){
      node->next = nextNode->next;
    } else {
        node->next = NULL;
    }
  }
};
